wgsl/index.bs

```diff
@@ -1266,8 +1266,10 @@ We say the identifier is <dfn noexport>in scope</dfn>
 
 Where a declaration appears determines its scope:
 * Predeclared objects, and objects declared at module-scope, are [=in scope=] across the entire program source.
+* Each [=formal parameter=] of a user-declared function is [=in scope=] across the corresponding [=function body=].
```

Comment:
It is strange that this instance says across the function body and later it just says function body. Just the body seems insufficient or the following would be allowed (which seems undesirable):
```
fn foo(param : i32, param : f32) { }
```

Easier to just say in scope until the end of the function body in both locations.  Maybe it would be better to refactor as:
>* The scope of each is a span of text beginning immediately after the end of the declaration.
>    * Each [=formal parameter=] of a user-declared function is [=in scope=] until the end of the [=function body=]. See [[#function-declaration-sec]].
>    * For other data declarations, see [[#var-and-value]].

Reply:
There is already a rule at https://gpuweb.github.io/gpuweb/wgsl/#formal-parameter :

> Two formal parameters for a given function [must not](https://gpuweb.github.io/gpuweb/wgsl/#shader-creation-error) have the same name.

Reply:
Interestingly, the rules allow this. They did before, and they still do:

      fn foo(foo: i32) { }

Reply:
About the "across the function body", the "across" bit is trying to be in parallel structure as the previous bullet point about module-scope declarations.

Would be happy to rephrase.  I considered this but felt something essential was missing:

> The [=scope=] of each formal parameter is the associated [=function body=].

This seems awkward.
> The [=scope=] of each [=formal parameter=] in a function declaration is the associated [=function body=].

Reply:
What about my suggested refactoring? Then we could drop the rule about no duplicate names since it would be covered by the end scope restrictions.

Reply:
> Then we could drop the rule about no duplicate names since it would be covered by the end scope restrictions.

Your refactoring would reject this program, when it shouldn't.  'b' and the return type are spelled 'i32' but refer to the name of the first formal parameter

        fn foo(i32: i32, b: i32) -> i32 { return i32; }
                            ^ invalid type

Reply:
Is this allowed?
```
fn foo() {
  let i32 : i32 = 0;
  let x : i32 = 0;
}
```

If not I'm not sure if we need parameters to support it. If it is allowed, I'm not sure why this is different.

Reply:
In the 2023-01-10 meeting the group agreed to take the intent of the change.